src/pudl/transform/ferc1.py

```diff
@@ -853,6 +1024,85 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
         )
 
 
+class PlantInServiceFerc1TableTransformer(Ferc1AbstractTableTransformer):
+    """A transformer for the Electric Plant in Service table."""
+
+    table_id: Ferc1TableId = Ferc1TableId.PLANT_IN_SERVICE_FERC1
+
+    @cache_df(key="dbf")
+    def align_row_numbers_dbf(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Align historical FERC1 DBF row numbers with XBRL account IDs."""
+        return pd.merge(
+            df,
+            get_ferc1_dbf_xbrl_glue(self.source_table_id(Ferc1Source.DBF)),
+            on=["report_year", "row_number"],
+        ).rename(columns={"xbrl_column_stem": "ferc_account_id"})
+
+    @cache_df("reshape_instant")
+    def process_instant_xbrl(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Convert row-oriented end-of-year balances to starting/ending balance cols."""
+        df["year"] = pd.to_datetime(df["date"]).dt.year
+        df.loc[df.report_year == (df.year + 1), "balance_type"] = "starting_balance"
+        df.loc[df.report_year == df.year, "balance_type"] = "ending_balance"
+        # If there are other years something is wrong.
+        assert df.balance_type.notna().all()
+        df = (
+            df.drop(["year", "date"], axis="columns")
+            .set_index(["entity_id", "report_year", "balance_type"])
+            .unstack()
+        )
+        df.columns = ["_".join(items) for items in df.columns.to_flat_index()]
+        return df.reset_index()
+
+    def wide_to_tidy_xbrl(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Reshape wide-format table based on FERC account column labels."""
+        df = df.drop(["start_date", "end_date"], axis="columns")
+        suffixes = [
+            "starting_balance",
+            "additions",
+            "retirements",
+            "transfers",
+            "adjustments",
+            "ending_balance",
+        ]
+        new_cols = []
+        for col in df.columns:
+            for suffix in suffixes:
+                if col.endswith(suffix):
+                    new_cols.append((suffix, re.sub(f"_{suffix}", "", col)))
+        new_col_idx = pd.MultiIndex.from_tuples(
+            new_cols, names=["value_type", "ferc_account_id"]
+        )
+
+        df = df.set_index(["entity_id", "report_year"])
+        df.columns = new_col_idx
+        df = (
+            df.stack(level=1)
+            .loc[
+                :,
+                [
+                    "starting_balance",
+                    "additions",
+                    "retirements",
+                    "transfers",
+                    "adjustments",
+                    "ending_balance",
+                ],
+            ]
+            .reset_index()
+        )
+        df.columns.name = None
+        return df
+
+    @cache_df("main")
+    def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Main transform.
+
+        Dummy for now.
+        """
+        return df
```

Comment:
this obviously needs updating if we are to integrate the table pre-sign or ferc acct num.

Reply:
Made it less cursory. We really don't have any transforms to do in here though yet. Might apply sign conventions here, or at least temporarily add on the categorical columns for aggregation.